Flatten fields to avoid type conflicts in ElasticSearch

arybczak opened the pull request:
With keys `time` and `status` it's far too easy to introduce type conflicts with other log messages in the application when logging to Elasticsearch.

jonathanjouty commented on src/Network/Wai/Log/Options.hs:
```diff
@@ -93,17 +93,15 @@ defaultLogRequest uuid req =
+    , "response_code"    .= statusCode (responseStatus resp)
+    , "response_message" .= ts (statusMessage (responseStatus resp))
+    , "full_time"        .= full time
+    , "processing_time"  .= processing time
```
_Nitpick:_ `time_full` and `time_processing` instead?
Not sure if that makes it easier to search for the field in ES/Kibana.

jonathanjouty commented on src/Network/Wai/Log/Options.hs:
```diff
@@ -93,17 +93,15 @@ defaultLogRequest uuid req =
-    , "time"   .= object [ "full"    .= full time
-                         , "process" .= processing time
-                         ]
+defaultLogResponse uuid req resp responseInfo time =
```
_Idea:_ as an alternative add a new `defaultLogResponseFlat` or something along those lines?
Then this could technically be a minor release: https://pvp.haskell.org/#decision-tree

wwwater replied:
Agree with Jonathan that having it as an alternative default would save us from reacting to it as a breaking change.

arybczak replied:
Is that a problem that the default changes?

wwwater replied:
We use it in the eID service, and we'll need to adjust a few graphs on the dashboards. Not a big problem, though. How does having these fields nested result in type conflicts?

jonathanjouty reviewed (approved):
Hi! 👋🏽   👻 

If you want to use this soon, maybe update the version in this PR so you can do a release?

Also any internal users, such as @wwwater, should be made aware of the change as it might affect dashboards they have configured 😉

arybczak commented:
I updated the PR and added a description. It would be good to merge this now since #8 also changed `defaultLogResponse`.

@wwwater are you fine with merging this? Kontrakcja already uses these settings and it would be good to have this as a default so all applications have the same structure of these log messages.

wwwater commented:
Yep, it's ok to merge.
